### ISSUE TITLE
Foundational Components of Neural Networks in PyTorch

### ISSUE THREAD

**Issue by vochicong** [vochicong/progress#183]
https://learning.oreilly.com/library/view/natural-language-processing/9781491978221/ch03.html#foundational_components_of_neural_networ

PyTorchのNN作成・学習は、Kerasよりも細かく記述するが、インタフェースがスッキリしている
- NNの定義 (Kerasに近い見た目）
- zero_grad() でグラディエントを消去
- NNにデータを通して、その結果をラベルと比較して損失を計算
- 損失からグラディエントを計算 loss.backward()
- 最適化機の optimizer.step() でNNのパラメーターを更新
